### PR TITLE
feat(neon_talk): Display references inside chat messages

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/1930
Depends on https://github.com/nextcloud/neon/pull/2116

Ready for review, just need the other PR to be merged first.

### CODE REVIEW

File: packages/neon/neon_talk/lib/src/widgets/message.dart [hunk 1]
```diff
@@ -356,24 +362,94 @@ class TalkCommentMessage extends StatelessWidget {
...
+        setState(() {
+          references = references.rebuild((b) {
+            for (final url in references.keys) {
+              b[url] = result[url] ?? Result.loading();
+            }
+
+            b.removeWhere((key, value) => !result.keys.contains(key));
+          });
+        });
+      });
```
Comment (Leptopoda) [inline packages/neon/neon_talk/lib/src/widgets/message.dart:424]: I think that this logic should be in the bloc

File: packages/neon_framework/lib/src/router.dart
Comment (Leptopoda) [inline packages/neon_framework/lib/src/router.dart:1]: This fix should have a test

File: packages/neon/neon_talk/lib/src/widgets/message.dart (hunk 1 above)
Comment (provokateurin) [reply packages/neon/neon_talk/lib/src/widgets/message.dart:424]: I wanted to do that at first, but it doesn't fit in the ReferencesBloc. I could make a bloc per message, but I'm not sure if that is a good idea.

**Comment by provokateurin** [comment]
The links inside the chat message should also be highlighted and clickable

**Review by Leptopoda — APPROVED** [review]

**Comment by provokateurin** [comment]
Ah I remember I wanted to cleanup the design of the reference previews, I forgot to do that.

**Comment by provokateurin** [comment]
@Leptopoda please re-review the latest changes I did to the design.

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**